### PR TITLE
Update & adjust Infection configuration

### LINKED ISSUE #1286: Revisit mutation tests
`infection.json` seems to not have all directories correctly configured so it's probably not even going through libraries and bridges. 
Including everything might reduce the MSI and coverage

### PR DESCRIPTION
**Pull request by stloyd**
<h2>Change Log</h2>
<div id="change-log">
  <h4>Added</h4>
  <ul id="added">
    
  </ul> 
  <h4>Fixed</h4>  
  <ul id="fixed">
    
  </ul>
  <h4>Changed</h4>
  <ul id="changed">
    <li>Update & adjust Infection configuration</li>
  </ul>  
  <h4>Removed</h4>
  <ul id="removed">
    
  </ul>
  <h4>Deprecated</h4>
  <ul id="deprecated">
    
  </ul>  
  <h4>Security</h4> 
  <ul id="security">
    
  </ul>     
</div>
<hr/>

<h2>Description</h2>

Fixes #1286

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -44,7 +44,7 @@
         "cocur/slugify": "^4.5",
         "fakerphp/faker": "^1.23",
         "fig/log-test": "^1.1",
-        "nikic/php-parser": "^4.18",
+        "nikic/php-parser": "^5.3",
```
Comment (stloyd) [inline composer.json:48]: Required by Infection, due too: https://github.com/infection/infection/issues/1984

File: infection.json [hunk 2]
```diff
@@ -42,8 +32,6 @@
     },
     "Throw_": {
       "ignore": [
-        "Flow\\ETL\\Adapter\\Doctrine\\DbalBulkLoader::__serialize",
-        "Flow\\ETL\\Adapter\\Doctrine\\DbalBulkLoader::__unserialize",
```
Comment (stloyd) [inline infection.json:46]: Dead code.

File: infection.json [hunk 3]
```diff
@@ -82,6 +70,6 @@
   },
   "tmpDir": "var/infection/cache",
   "testFrameworkOptions": "--testsuite=unit",
-  "minMsi": 40,
+  "minMsi": 25,
```
Comment (stloyd) [inline infection.json:73]: Currently we don't have such MSI...

File: .github/workflows/test-mutations.yml [hunk 4]
```diff
@@ -78,3 +78,12 @@ jobs:
...
         run: "composer test:mutation -- --logger-github=false"
+
+      - name: "Store Infection logs"
+        if: success() || failure()
+        uses: actions/upload-artifact@v4
+        with:
+          name: infection-logs-pr-${{ github.event.pull_request.number }}
+          path: ./var/infection/*.log
+          overwrite: true
+          retention-days: 7
```
Comment (norberttech) [inline .github/workflows/test-mutations.yml]: if those logs are readable we can add them to job summary: 

```
      - name: "Set job summary"
        run: |
          cat ./var/infection/*.log >> $GITHUB_STEP_SUMMARY
```

Comment (stloyd) [reply .github/workflows/test-mutations.yml]: Main is... 7MB+ of data 😅 but summary is not best either, but after checking them I'm not sure it's worth holding them anyway

Comment (norberttech) [reply .github/workflows/test-mutations.yml]: oh shit, then maybe lets drop it 😅

**Pull request merged**